library/ssl_msg.c

```diff
@@ -1613,6 +1613,9 @@ int mbedtls_ssl_decrypt_buf( mbedtls_ssl_context const *ssl,
         else
 #endif /* MBEDTLS_SSL_PROTO_TLS1 || MBEDTLS_SSL_PROTO_TLS1_1 || \
           MBEDTLS_SSL_PROTO_TLS1_2 */
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+        if( transform->minor_ver != MBEDTLS_SSL_MINOR_VERSION_4 )
```

Comment:
@oesh The present code-path is for CBC mode encryption, which isn't supported in TLS 1.3. It is therefore an error if we get to this point if `transform->minor_ver == MBEDTLS_SSL_MINOR_VERSION_4`, and it seems that the previous version of the code was right. What was the motivation for the change?

Reply:
How about guarding the entire CBC codepath with `#if defined( SSL3 || TLS1 || TLS1_1 || TLS1_2 ) ... #endif`? That should resolve the issue.